content/docs/graphql/read-only-tokens.md

```diff
@@ -39,9 +43,94 @@ Wild card matching is supported in the branch names using '\*' to match anything
 
 Wild card matching is useful for matching branches that have not been created yet and can be used for editorial workflows.
 
-### Make a fetch request using the API key
+### Making requests with the tina client
+#### Setting up the Tina client
+
+This client can be used for data fetching with Tina. It can be used on the client and server to make graphQL requests. 
+
+To set up first add a new file called `.tina/client.{js,ts}`
+
+```ts
+const branch = "main";
+const apiURL =
+  process.env.NODE_ENV == "development"
+    ? "http://localhost:4001/graphql"
+    : `https://content.tinajs.io/content/<Your Client ID>/github/${branch}`;
+
+// Token generated on app.tina.io
+export const client = createClient({
+    // default values
+    url: apiURL,
+    token: "Your Read Only Token generated above",
+})
+```
+When using "http://localhost:4001/graphql" the content will be queried from file system (This only works during devolvement or in CI) and when using `https://content.tinajs.io/content/<Your Client ID>/github/${branch}` the content will be queried from Tina cloud. 
+
+In most cases the `apiURL` is the same one that is used for editing. So instead of passing the passing `apiURL` the client can be passed.
 
-Now you can make a POST request to the content API with the desired GraphQL request.
+`.tina/schema.{ts,tsx,js}`
+
+```diff
+import { client } from './client'
+//...
+export const tinaConfig = defineConfig({
++  client,
+-  url: ...
+  //...
+})
+```
+
+#### Requesting data using Read only tokens
+
+Now that the client is defined we can use it to query our content.
+
+```ts
+import { client } from '../pathToTina/.tina/client'
+
+const data = await client.request({query: "Your Graphql query"})
+```
+
+`variables` can also be passed
+
+```ts
+import { client } from "../pathToTina/.tina/client";
+
+const data = await client.request({
+  query: "Your Graphql query",
+  variables: {...},
+});
+
+```
+
+*optionally* the `url` or `token` can be overridden from request to request.
```

Comment:
Is the only case where we can see this being overridden only for static with the filesystem? I almost wonder if this belongs more in a reference section.

Reply:
I reference section would be good for this. Only reason I did not include it was because I did not want users to see that before reading this doc.